src/app/modules/auth/auth.controller.ts

```diff
@@ -8,167 +7,223 @@ import { createLoggerWithLabel } from '../../../config/logger'
 import { LINKS } from '../../../shared/constants'
 import MailService from '../../services/mail.service'
 import { getRequestIp } from '../../utils/request'
-import { ApplicationError } from '../core/core.errors'
+import { ApplicationError, DatabaseError } from '../core/core.errors'
+import { MailSendError } from '../mail/mail.errors'
 import * as UserService from '../user/user.service'
 
+import { InvalidDomainError, InvalidOtpError } from './auth.errors'
 import * as AuthService from './auth.service'
-import { ResponseAfter, SessionUser } from './auth.types'
+import { SessionUser } from './auth.types'
 
 const logger = createLoggerWithLabel(module)
 
 /**
- * Precondition: AuthMiddlewares.validateDomain must precede this handler.
- * @returns 200 regardless, assumed to have passed domain validation.
+ * Handler to map ApplicationErrors to their correct status code and error
+ * messages.
+ * @param error The error to retrieve the status codes and error messages
+ * @param coreErrorMessage Any error message to return instead of the default core error message, if any
  */
-export const handleCheckUser: RequestHandler = async (
-  _req: Request,
-  res: ResponseAfter['validateDomain'],
-) => {
-  return res.sendStatus(StatusCodes.OK)
+const mapRouteError = (error: ApplicationError, coreErrorMessage?: string) => {
```

Comment:
I think each `*.controller` file should have it's own routing function, since each controller should only handle a small subset of errors. Do you think that is reasonable to assume? 

Or did you mean to extract to `auth.utils` instead of a core utility file

Reply:
Yup the latter

Reply:
I agree that each controller should have its own routing function

Reply:
Done in 05d3f58